### PR TITLE
Kmd client

### PR DESCRIPTION
**Pull request by amityadav0**

### CODE REVIEW

File: docs/algob-config.md [hunk 1]
```diff
@@ -24,21 +24,21 @@ initialDate?: string;
...
 1. Initializing a native `algosdk.Account` object. Example:
 
         const account = {
-          name: 'master'
+          name: 'master',
           addr: 'UDF7DS5QXECBUEDF3GZVHHLXDRJOVTGR7EORYGDBPJ2FNB5D5T636QMWZY',
           sk: new Uint8Array([28,  45,  45,  15,  70, 188,  57, 228,  18,  21,  42, 228,  33, 187, 222, 162,  89,  15,  22,  52, 143, 171, 182,  17, 168, 238,  96, 177,  12, 163, 243, 231, 160, 203, 241, 203, 176, 185,   4,  26,  16, 101, 217, 179, 83, 157, 119,  28,  82, 234, 204, 209, 249,  29,  28, 24,  97, 122, 116,  86, 135, 163, 236, 253]) }
 
-1. Loaded from a file (you can generate a test accounts using `gen-accounts` command). `algob` has to be available in your node_modules.
+1. Loading from a file (you can generate a test accounts using `gen-accounts` command). `algob` has to be available in your node_modules.
```
Comment (ratik21) [inline docs/algob-config.md:36]:
```suggestion
1. Loading from a file (you can generate a test accounts using `gen-accounts` command). `algob` should be available in your node_modules.
```

File: packages/algob/sample-project/algob.config.js [hunk 2]
```diff
@@ -5,49 +5,42 @@
 // NOTE: to be able to execute transactions, you need to use an active account with
 // a sufficient ALGO balance.
 
+/**
+  Check our [algob-config.md](/docs/algob-config.md) documentation for more ways how to
+  load a private keys:
```
Comment (ratik21) [inline packages/algob/sample-project/algob.config.js:10]:
```suggestion
  load a private key:
```

File: packages/algob/sample-project/algob.config.js [hunk 3]
```diff
@@ -5,49 +5,42 @@
...
-  sk: new Uint8Array([28,  45,  45,  15,  70, 188,  57, 228,  18,  21,  42, 228,  33, 187, 222, 162,  89,  15,  22,  52, 143, 171, 182,  17, 168, 238,  96, 177,  12, 163, 243, 231, 160, 203, 241, 203, 176, 185,   4,  26,  16, 101, 217, 179, 83, 157, 119,  28,  82, 234, 204, 209, 249,  29,  28, 24,  97, 122, 116,  86, 135, 163, 236, 253])
-}]
-
 
 let defaultCfg = {
   host: "http://localhost",
   port: 8080,
   token: "content_of/algorand-node-data/algod.token",
   accounts: accounts,
+  // if you want to load accoutns from KMD, you need to add the kmdCfg object. Please read
```
Comment (ratik21) [inline packages/algob/sample-project/algob.config.js:38]:
```suggestion
  // if you want to load accounts from KMD, you need to add the kmdCfg object. Please read
```

File: packages/algob/sample-project/algob.config.js [hunk 4]
```diff
@@ -5,49 +5,42 @@
...
+  // algob-confg.md documentation for details.
```
Comment (ratik21) [inline packages/algob/sample-project/algob.config.js:39]:
```suggestion
  // algob-config.md documentation for details.
```

File: examples/deployment/algob.config.js [hunk 5]
```diff
@@ -67,11 +77,14 @@ let defaultCfg = {
   port: 46469,
   token: "aade468d25a7aa48fec8082d6a847c48492066a2741f3731e613fdde086cd6e9",
   accounts: accounts,
+  kmdConfig: kmdconfig,
+  kmdAddresses: await loadKMDAccounts(kmdconfig.host, kmdconfig.token, kmdconfig.port, 
+    kmdconfig.walletname, kmdconfig.walletpassword)
```
Comment (amityadav0) [inline examples/deployment/algob.config.js:82]: can we pass it `loadKMDAccounts(kmdconfig)` like this ?

File: packages/algob/src/internal/core/config/config-loading.ts [hunk 6]
```diff
@@ -38,18 +39,40 @@ export function loadConfigAndTasks (
...
+  const netname = runtimeArgs?.network;
+  if (netname !== undefined) {
+    const net = cfg.networks[netname];
+    if (net !== undefined) { await _loadKMDAccounts(net); }
+  }
+
+  return cfg;
+}
+
+// loads KMD accoutns if the net.kmdCfg is specified and merges them into net.accounts
```
Comment (ratik21) [inline packages/algob/src/internal/core/config/config-loading.ts:65]:
```suggestion
// loads KMD accounts if net.kmdCfg is specified and merges them into net.accounts
```

File: packages/algob/sample-project/algob.config.js [hunk 7]
```diff
@@ -5,49 +5,42 @@
...
-// let accounts = mkAccounts([{
-//   addr: "KFMPC5QWM3SC54X7UWUW6OSDOIT3H3YA5UOCUAE2ABERXYSKZS5Q3X5IZY",
-//   mnemonic: "call boy rubber fashion arch day capable one sweet skate outside purse six early learn tuition eagle love breeze pizza loud today popular able divide"
-// }]);
-//
-// To see your algob account and extract menemonic run:
-// 	goal -d $(ALGORAND_DATA) account list
-// 	goal -d $(ALGORAND_DATA) account export -a <account address>
+const { mkAccounts } = require("algob");
+let accounts = mkAccounts([{
```
Comment (amityadav0) [inline packages/algob/sample-project/algob.config.js:21]: this config also needs to have async loading `module.exports = (async function() {` else it fails
giving this error.

File: packages/algob/sample-project/algob.config.js [hunk 8]
```diff
@@ -5,49 +5,42 @@
...
+  addr: "KFMPC5QWM3SC54X7UWUW6OSDOIT3H3YA5UOCUAE2ABERXYSKZS5Q3X5IZY",
+  mnemonic: "call boy rubber fashion arch day capable one sweet skate outside purse six early learn tuition eagle love breeze pizza loud today popular able divide"
```
Comment (amityadav0) [inline packages/algob/sample-project/algob.config.js:23]: `name` field is missing from here (for accounts)

File: packages/algob/sample-project/algob.config.js (hunk 7 above)
Comment (amityadav0) [reply packages/algob/sample-project/algob.config.js:21]: i think this needs to be updated for all example projects, should i update it ?

Comment (amityadav0) [inline packages/algob/sample-project/algob.config.js:21]:
```suggestion
module.exports = (async function() {
let accounts = mkAccounts([{
```

File: examples/deployment/algob.config.js (hunk 5 above)
Comment (robert-zaremba) [reply examples/deployment/algob.config.js:82]: No, this is wrong. It shouldn't be here. I'm removing this.

File: packages/algob/sample-project/algob.config.js (hunk 2 above)
Comment (robert-zaremba) [reply packages/algob/sample-project/algob.config.js:10]: The original form is correct.

File: packages/algob/sample-project/algob.config.js (hunk 7 above)
Comment (robert-zaremba) [reply packages/algob/sample-project/algob.config.js:21]: Thanks for catching. I want to avoid `async` in config. We shouldn't use it here.

Comment (robert-zaremba) [reply packages/algob/sample-project/algob.config.js:21]: The problem was in the exports, I forgot to add a `networks` layer.

File: packages/algob/sample-project/algob.config.js (hunk 8 above)
Comment (robert-zaremba) [reply packages/algob/sample-project/algob.config.js:23]: yes, I've updated it in the `project-dev` and forgot to update here.

**Comment by amityadav0** [comment]
Looks good.

**Review by ratik21 — APPROVED** [review]

**Comment by robert-zaremba** [comment]
This got merged manually

**Pull request closed without merging**